src/input-states.js

```diff
@@ -0,0 +1,256 @@
+/**
+ * input-states.js
+ * 
+ * Manages RDF pipeline input states for a VNI
+ * TODO: Add optimization for input states as a sorted array
+ */
+
+var _ = require('underscore');
+
+var InputType = {
+    IIP: 1,
+    PACKET: 2
+};
+var ALL_PORTINFO_KEYS = [ 'inportName', 'sourceNodeName', 'sourcePortName' ];
+
+var inputStates = [];
+
+/**
+ * Get the index to the state in the input states array if it exists already
+ * 
+ * @param portInfo an object that describes the port(s) associated with the data
+ *
+ * @return an integer index into the inputStates array if a state is found or -1 if no state was found.
+ */
+function getStateIndex( portInfo ) { 
+
+    var portSourceNodeName = portInfo.sourceNodeName || '';
+    var portSourcePortName = portInfo.sourcePortName || '';
+
+    return _.findIndex( inputStates, function(inputState) { 
+        
+        var stateSourceNodeName = inputState.sourceNodeName || '';
+        var stateSourcePortName = inputState.sourcePortName || '';
+
+        return ( portInfo.inportName === inputState.inportName &&
+                 portSourceNodeName === stateSourceNodeName &&
+                 portSourcePortName === stateSourcePortName );
+    });
+}
+
+
+function insertAtIndex( portInfo ) { 
```

Comment:
Suggest renaming this to insertionIndex, so that it does not sound like it is performing the actual insertion.